ui/html-fragment.reel/html-fragment.js

```diff
@@ -0,0 +1,223 @@
+/*global require, exports */
+
+var Component = require("ui/component").Component,
+    defaultOptions = Object.deepFreeze({
+        allowedTags: [
+            'h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'blockquote', 'p', 'a', 'ul', 'ol',
+            'nl', 'li', 'b', 'i', 'img', 'strong', 'em', 'strike', 'code', 'hr',
+            'br', 'div', 'table', 'thead', 'caption', 'tbody', 'tr', 'th', 'td',
+            'pre', 'span'
+        ],
+        allowedAttributes: {
+            '*': [
+                'href', 'align', 'alt', 'center', 'bgcolor', 'src', 'title',
+                'height', 'width', 'data-*', 'style'
+            ],
+            a: ['href', 'name', 'target'],
+            img: ['src']
+        }
+    });
+
+/**
+ *
+ * @class HtmlFragment
+ * @extends Component
+ */
+var HtmlFragment = exports.HtmlFragment = Component.specialize(/** @lends HtmlFragment# */ {
+
+    initWithOptions: {
+        value: function (options) {
+            this.options = options;
+        }
+    },
+
+    _value: {
+        value: null
+    },
+
+    value: {
+        set: function (value) {
+            if (this._value !== value) {
+                if (typeof value === 'string') {
+                    this._value = value;
+                } else {
+                    this._value = null;
+                }
+
+                this.needsSanitizeHtml = true;
+                this.needsDraw = true;
+            }
+        },
+        get: function () {
+            return this._value;
+        }
+    },
+
+    allowedTags: {
+        value: null
+    },
+
+    allowedAttributes: {
+        value: null
+    },
+
+    _getSanitizerOptions: {
```

Comment:
replace by 2 getters, one for allowedTags and one for allowedAttributes

Reply:
@marchant Having a function for getting the options could help us for future features, such as filtering css rules, replacing tag names...
 Moreover `_sanitizeHtml` is an internal and private method.